pilot/pkg/security/authz/model/model.go

```diff
@@ -265,3 +270,39 @@ func (m *Model) Generate(service *ServiceMetadata, forTCPFilter bool) *envoy_rba
 	}
 	return policy
 }
+
+// getPrincipalsIncludingAliases returns a list of users encoded in SPIFFE format with the local
+// trust domain and its aliases.
+// For example, for a user "bar" in namespace "foo".
+// If the local trust domain is "td2" and its alias is "td1" (migrating from td1 to td2),
+// getPrincipalsIncludingAliases returns ["td2/ns/foo/sa/bar", "td2/ns/foo/sa/bar]].
+func getPrincipalsIncludingAliases(trustDomain string, trustDomainAliases []string, principals []string) []string {
+	principalsIncludingAliases := []string{}
+	for _, principal := range principals {
+		trustDomainFromPrincipal := extractTrustDomainFromAuthzPrincipal(principal)
+		// If the current trust domain is the same as the one from the existing principal, there is nothing to do.
+		// NOTE: All |principals| must have the same trust domain.
+		if trustDomainFromPrincipal == trustDomain {
```

Comment:
Don't you still need to expand for aliases? If not, how can you allow request with the aliased principal (say td1)?

Reply:
I think the goal of this is for trust domain migration, so a typical use case would be setting the new trust domain to be different from the existing one from the policy. If they are the same (and trust domain aliases are not empty), it's most likely a mistake from the users?

Reply:
My understand of aliases is for merging multiple meshes as well. In that cases, you may have a workloads that uses legacy TD (`td1`). If all authZ policy already use new TD `td2`, based on this logic, requests from that workload will be rejected, right?

And even for simple migration on the same mesh, it's possible when some authZ policies already use the new TD2, but not all workloads migrate to that yet. So again, it will break traffic.

Reply:
The goal of the relevant design doc and this PR is for trust domain migration only. As for mesh federation, I think it's out of the PR's scope. 

For the second point, say we have an old cluster with td1. When migrating to td2, The users do not expect to edit the policy files (`trustDomainFromPrincipal` extracts the trust domain from the policy file), so it's not usual for the extracted trust domain to be the new trust domain, but I'll remove this check if it looks better/safer.

Reply:
Removed. Thanks!